src/txn/txn_recover.c

```diff
@@ -906,8 +961,10 @@ __wt_txn_recover(WT_SESSION_IMPL *session, const char *cfg[])
      * Recovery can touch more data than fits in cache, so it relies on regular eviction to manage
      * paging. Start eviction threads for recovery without history store cursors.
      */
-    WT_ERR(__wt_evict_create(session));
-    eviction_started = true;
+    if (!eviction_started) {
```

Comment:
Can you summarise the changes in this file? It's hard to tell by looking at the code + the PR description what we're trying to achieve here.

On another note, the control flow of this function is getting pretty complicated...

Reply:
The main changes here are to perform a rollback to stable before log recovery and because of which I introduced some jump blocks to keep the current code intact.

Reply:
Actually, why is this fine?

@kommiharibabu Doesn't that mean that unstable updates that were logged will remain after we've started up?

Reply:
@tetsuo-cpp Logged tables are recovered till the last transaction commits before the crash. RTS doesn't do anything on logged tables. So moving the RTS before log replay doesn't have any problem as earlier also these tables are anyway ignored during RTS.